packages/ui/src/components/SelectableCardGroup/index.tsx

```diff
@@ -0,0 +1,171 @@
+  showTick: boolean
+} & Required<Pick<InputHTMLAttributes<HTMLInputElement>, 'onChange'>> &
+  Pick<InputHTMLAttributes<HTMLInputElement>, 'required'>
+
+const SelectableCardGroupContext = createContext<
+  SelectableCardGroupContextType | undefined
+>(undefined)
+
+type CardSelectableCardProps = Omit<
+  ComponentProps<typeof SelectableCard>,
+  'onChange' | 'checked' | 'type' | 'showTick'
+>
+
+const CardSelectableCard = ({
+  value,
+  disabled,
+  children,
+  className,
+  isError,
+  onFocus,
+  onBlur,
+  tooltip,
+  id,
+  label,
+  'data-testid': dataTestId,
+}: CardSelectableCardProps) => {
+  const context = useContext(SelectableCardGroupContext)
+
+  if (!context) {
+    throw new Error(
+      'SelectableCardGroup.Card can only be used inside a SelectableCardGroup',
+    )
+  }
+
+  const { groupName, onChange, groupValue, type, showTick } = context
+
+  return (
+    <SelectableCard
+      name={groupName}
+      value={value}
+      showTick={showTick}
+      type={type}
+      disabled={disabled}
+      className={className}
+      isError={isError}
+      onFocus={onFocus}
+      onBlur={onBlur}
+      tooltip={tooltip}
+      id={id}
+      label={label}
+      onChange={onChange}
+      checked={
+        typeof groupValue === 'object'
+          ? groupValue.includes(value)
+          : groupValue === value
+      }
+      data-testid={dataTestId}
+    >
+      {children}
+    </SelectableCard>
+  )
+}
+
+const FieldSet = styled.fieldset`
+  border: none;
+  padding: 0;
+  margin: 0;
+`
+
+const StyledRequiredIcon = styled(Icon)`
+  vertical-align: super;
+`
+
+type SelectableCardGroupProps = {
+  legend: string
+  value: string | number | (string | number)[]
+  className?: string
+  helper?: ReactNode
+  error?: ReactNode
+  columns?: number
+  children: ReactNode
+  type: 'radio' | 'checkbox'
+  required?: boolean
+  showTick?: boolean
+} & Required<Pick<InputHTMLAttributes<HTMLInputElement>, 'onChange' | 'name'>>
+
+/**
+ * SelectableCardGroup is a component that allows users to select cards from a list of cards using SelectableCard.
+ */
+export const SelectableCardGroup = ({
+  legend,
+  value,
+  className,
+  helper,
+  error,
+  columns = 1,
+  children,
+  onChange,
+  name,
+  required = false,
+  type,
+  showTick = false,
+}: SelectableCardGroupProps) => {
+  const contextValue = useMemo(
+    () => ({
+      groupName: name,
+      groupValue: value,
+      onChange,
+      required,
+      type,
+      showTick,
+    }),
+    [name, value, onChange, required, type, showTick],
+  )
+
+  return (
+    <SelectableCardGroupContext.Provider value={contextValue}>
+      <Stack gap={1}>
+        <FieldSet className={className}>
+          <Stack gap={1.5}>
+            <Text as="legend" variant="bodyStrong">
+              {legend}&nbsp;
+              {required ? (
+                <StyledRequiredIcon name="asterisk" color="danger" size={8} />
+              ) : null}
+            </Text>
```

Comment:
Here you can add a condition to check if legend exists or not and do not render it if it doesn't